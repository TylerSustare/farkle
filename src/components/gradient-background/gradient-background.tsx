import { LinearGradient } from 'expo-linear-gradient';
import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { View } from 'react-native';

interface Props {
  children: React.ReactNode;
}

export function GradientBackground({ children }: Props): React.ReactElement {
  return (
    <View style={{ flex: 1 }}>
      <StatusBar style="light" />
      <LinearGradient
        colors={['#006ea1', '#cf2b2b']}
        style={{
          position: 'absolute',
          left: 0,
          right: 0,
          top: 0,
          bottom: 0,
        }}
      />
      {children}
    </View>
  );
}
