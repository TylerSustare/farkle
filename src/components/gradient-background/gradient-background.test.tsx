import { cleanup, fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Button, Text, View } from 'react-native';
import { GradientBackground } from './gradient-background';

describe('<GradientBackground/>', () => {
  afterEach(cleanup);
  it('should render a <GradientBackground /> with children', () => {
    const { container, toJSON } = render(
      <GradientBackground>
        <View>
          <Text>Hello</Text>
        </View>
      </GradientBackground>,
    );
    expect(container).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should not modify children components in <GradientBackground />', () => {
    const fn = jest.fn();
    const { getByText } = render(
      <GradientBackground>
        <Button title="Hello" onPress={fn} />
      </GradientBackground>,
    );
    const button = getByText('Hello');
    fireEvent.press(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
