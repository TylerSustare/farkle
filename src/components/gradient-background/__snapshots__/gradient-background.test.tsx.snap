// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientBackground/> should render a <GradientBackground /> with children 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278218401,
          4291767083,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View>
    <Text>
      Hello
    </Text>
  </View>
</View>
`;
